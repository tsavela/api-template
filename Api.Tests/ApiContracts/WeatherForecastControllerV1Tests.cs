using System.Net;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Api.Tests.Setup;
using ApprovalTests;
using FluentAssertions;
using Microsoft.AspNetCore.TestHost;
using Xunit;
using Xunit.Abstractions;

namespace Api.Tests.ApiContracts
{
    /// <summary>
    /// The purpose with these tests are to assert that the API contract is still in place,
    /// not to break the API for any consumers.
    /// </summary>
    public class WeatherForecastControllerV1Tests
    {
        private readonly TestServer _testServer;

        public WeatherForecastControllerV1Tests(ITestOutputHelper output)
        {
            _testServer = new TestServerBuilder(output).Build();
        }

        [Fact]
        public async Task GetWeatherForecast_ShouldReturnJsonAccordingToContract()
        {
            // Important to use a fixed json string and not serialize a request type, to avoid accidentally
            // changing the request when refacting, breaking the contract without noticing.
            var body = "{\"forecastLengthInDays\": 3}";
            var client = _testServer.CreateClient();

            var actual = await client.PostAsync("/api/v1/WeatherForecast",
                new StringContent(body, Encoding.UTF8, "application/json"));

            actual.StatusCode.Should().Be(HttpStatusCode.OK);

            Approvals.Verify((await actual.Content.ReadAsStringAsync()).PrettifyJson());
        }
    }
}
