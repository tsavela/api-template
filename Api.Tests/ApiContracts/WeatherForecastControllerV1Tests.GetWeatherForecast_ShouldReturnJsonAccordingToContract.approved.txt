{
  "forecasts": [
    {
      "daysFromNow": 1,
      "temperatureC": 5,
      "summary": "Freezing"
    },
    {
      "daysFromNow": 2,
      "temperatureC": 10,
      "summary": "Bracing"
    },
    {
      "daysFromNow": 3,
      "temperatureC": 15,
      "summary": "Chilly"
    }
  ]
}